pkg/api/types.go

```diff
@@ -759,6 +767,11 @@ func (l *LinuxProfile) HasSecrets() bool {
 	return len(l.Secrets) > 0
 }
 
+// HasSearchDomain returns true if the customer specified secrets to install
+func (l *LinuxProfile) HasSearchDomain() bool {
+	return l.CustomSearchDomain != nil
```

Comment:
Maybe this should check the content (Name, RealmUser and Password) to make sure that they are not empty? other wise if my apimodel looks like: 

```

"linuxProfile": { 


 
+        "customSearchDomain": {}, 


 
+        "adminUsername": "azureuser", 


 
+        "ssh": { 


 
+          "publicKeys": [ 


 
+            { 


 
+              "keyData": "" 


 
+            } 


 
+          ] 


 
+        } 


 
+      }, 
```
acs-engine will try to apply the script. That or we shouls enforce validation to make sure that if CustomSearchDomain is specified it has a valid Name + User + Password combination.